### PR TITLE
Minor updates to the AD page.

### PR DESCRIPTION
**Pull request by bangerth**
Just trying to educate myself about this page.

### CODE REVIEW

File: doc/doxygen/headers/automatic_and_symbolic_differentiation.h [hunk 1]
```diff
@@ -20,8 +20,34 @@
...
  * Below we provide a very brief introduction as to what automatic and symbolic differentiation are,
- * what variations of these computational / numerical schemes exist, and how they are integrated
- * within deal.II's framework.
+ * what variations of these computational/numerical schemes exist, and how they are integrated
+ * within deal.II's framework. The purpose of all of these schemes is to automatically compute
+ * (approximations of) the derivative of functions in cases where one does not want to
+ * compute them by hand. Common examples are situations in the finite element context is where
+ * one wants to solve a nonlinear problem that is given by requiring that some residual
+ * $F(u,\nabla u)=0$ where $F$ is a complicated function that needs to be differentiated to
+ * apply a Newton method; and situations where one is given a parameter dependent problem
```
Comment (jppelteret) [inline doc/doxygen/headers/automatic_and_symbolic_differentiation.h]: Maybe this should be "apply Newton's method" or "apply a Newton-like method". I suppose there is only one "Newton–Raphson method".

File: doc/doxygen/headers/automatic_and_symbolic_differentiation.h [hunk 2]
```diff
@@ -20,8 +20,34 @@
  * to automatic and symbolic differentiation.
  *
  * Below we provide a very brief introduction as to what automatic and symbolic differentiation are,
- * what variations of these computational / numerical schemes exist, and how they are integrated
- * within deal.II's framework.
+ * what variations of these computational/numerical schemes exist, and how they are integrated
+ * within deal.II's framework. The purpose of all of these schemes is to automatically compute
+ * (approximations of) the derivative of functions in cases where one does not want to
```
Comment (jppelteret) [inline doc/doxygen/headers/automatic_and_symbolic_differentiation.h]: I would prefer this to read something like

> The purpose of all of these schemes is to automatically compute the derivative of functions, or approximations thereof, ...

As is stated later, well-formulated AD method (and, by nature, correctly implemented SD methods) may return exact derivatives to machine precision for first and sometimes higher-order derivatives.

File: doc/doxygen/headers/automatic_and_symbolic_differentiation.h [hunk 3]
```diff
@@ -20,8 +20,34 @@
...
+ * compute them by hand. Common examples are situations in the finite element context is where
+ * one wants to solve a nonlinear problem that is given by requiring that some residual
+ * $F(u,\nabla u)=0$ where $F$ is a complicated function that needs to be differentiated to
+ * apply a Newton method; and situations where one is given a parameter dependent problem
+ * ${\cal A}(q,u,\nabla u) = f$ and wants to form derivatives with regards to $q$, for example
+ * to optimize an output functional with regards to $q$, or for a sensitivity analysis with
```
Comment (jppelteret) [inline doc/doxygen/headers/automatic_and_symbolic_differentiation.h]: What does "q" represent here? I presume its the "parameter" (material, geometric or something else?) that you refer to earlier?

File: doc/doxygen/headers/automatic_and_symbolic_differentiation.h [hunk 4]
```diff
@@ -20,8 +20,34 @@
...
+ * regards to $q$. In all of these cases, one should think of $F$ and $\cal A$ as <i>complicated</i>
+ * and cumbersome to differentiate -- at least when doing it by hand. A relatively simple case of
+ * a nonlinear problem that already shows the tedium of computing derivatives by hand is shown in
```
Comment (jppelteret) [inline doc/doxygen/headers/automatic_and_symbolic_differentiation.h]: already shows -> already highlights

(you use "show" twice in this sentence)

File: doc/doxygen/headers/automatic_and_symbolic_differentiation.h [hunk 5]
```diff
@@ -20,8 +20,34 @@
...
+ * temperature, and pressure of the fluid at each point; and where the chemical species react with
+ * each other based on reaction coefficients that also depend nonlinearly and in complicated
+ * ways on the chemical composition, temperature, and pressure. In many cases, the exact formulas
+ * for all of these coefficients can take several lines to write out, may include exponentials
+ * and (harmonic or geometric) averages of several nonlinear terms, and/or may contain table
+ * lookup of and interpolation between data points. Just getting these terms right is difficult
+ * enough; computing derivatives of these terms is impractical in most applications and, in
+ * reality, impossible to get right. Higher derivatives are even more impossible to do
+ * without computer aid. Automatic or symbolic differentiation is a way out of this:
+ * One only has to implement the coefficients only once, and gets the (correct!) derivatives without
```
Comment (jppelteret) [inline doc/doxygen/headers/automatic_and_symbolic_differentiation.h]: > One only has to implement the coefficients only once

Maybe you could clarify this a bit in the text. Do you mean that its only necessary to define a function for the coefficients in terms of all their dependencies?

File: doc/doxygen/headers/automatic_and_symbolic_differentiation.h [hunk 6]
```diff
@@ -113,7 +141,9 @@
  *    The outermost leaves on the tree represent the independent variables or constants, and are transformed by unary
  *    operators and connected by binary operators (in the most simple case). Therefore, the operations performed on
  *    the function inputs is known at compile time, and with that the associated derivative operation can also be defined
- *    at the same time. The compiled output type returned by this operator need not be generic, but can rather be
+ *    at the same time using the well-known rules of computing the derivative of an operation (such as
+ *    the associativity of derivative under addition and subtraction, the product rule, and the chain
```
Comment (jppelteret) [inline doc/doxygen/headers/automatic_and_symbolic_differentiation.h]: associativity of derivative -> associativity of derivatives

(plural)

**Comment by tamiko** [comment]
In reference to #8080, scheduled for rc3

**Comment by bangerth** [comment]
I pushed changes as requested. In particular I've expanded the part about the parameters `q` and given examples.

We can put this into 9.1, but it's really of no consequence. Nothing here is important. I'd much rather we now focus on only fixing bugs :-)

**Comment by jppelteret** [comment]
Thank you for considering my suggestions!

**Comment by jppelteret** [comment]
I'm merging since these are only documentation changes.

**Pull request merged**